src/main/java/org/opensearch/knn/index/vectorvalues/KNNFloatVectorValues.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.index.vectorvalues;
+
+import org.apache.lucene.codecs.KnnFieldVectorsWriter;
+import org.apache.lucene.index.BinaryDocValues;
+import org.apache.lucene.index.FloatVectorValues;
+import org.opensearch.knn.index.VectorDataType;
+
+import java.io.IOException;
+
+/**
+ * Concrete implementation of {@link KNNVectorValues} that returns float[] as vector and provides an abstraction over
+ * {@link BinaryDocValues}, {@link FloatVectorValues}, {@link KnnFieldVectorsWriter} etc.
+ */
+public class KNNFloatVectorValues extends KNNVectorValues<float[]> {
+    KNNFloatVectorValues(final KNNVectorValuesIterator vectorValuesIterator) {
+        super(vectorValuesIterator);
+    }
+
+    @Override
+    public float[] getVector() throws IOException {
+        final float[] vector = vectorValuesIterator.getVectorExtractorStrategy().extract(VectorDataType.FLOAT, vectorValuesIterator);
+        this.dimension = vector.length;
```

Comment:
since it was just a setting of int, I was not fond of making it lazily as it will add some branches. Hence I did it like this. Since for BDV, till you don't read the vector you don't know what is the dimension. It is coming from there

Reply:
@jmazanec15 I added some details on how to resolve this concern here: https://github.com/opensearch-project/k-NN/pull/1897#discussion_r1699570249 please let me know if this an acceptable approach here.